### PR TITLE
Adding initial proposal docs

### PR DESCRIPTION
**Pull request by johnewart**

### CODE REVIEW

File: templates/lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+- [ ] APIs that are used in the building block also meet Beta criteria
+
+
+## Stable 
+
+
+- [ ] Documentation is complete in `dapr/docs` with any changes since Beta
+- [ ] E2E scenarios well defined and comprehensive
+- [ ] Performance tests exist and regressions will prevent them from successfully passing
+- [ ] Performance data added to documentation
```
Comment (shubham1172) [inline templates/lifecycle.md:77]: Should this go here? https://docs.dapr.io/operations/performance-and-scalability/

File: templates/lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+- [ ] Interfaces to be used by `dapr/dapr` code defined and agreed upon
+- [ ] New building block package is defined in `components-contrib` repo
+- [ ] Conformance tests validating the components compliance added
+- [ ] Minimum of _one_ implementation (preferably something already in-use such as Redis if possible to reduce complexity)
+
+
+## Beta
+
+- [ ] E2E tests are up-to-date and comprehensive
```
Comment (artursouza) [inline templates/lifecycle.md:51]: Add SDK spec too

File: guides/api-design.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+APIs are expected to go through three stages in their lifetime: Alpha, Beta and Stable. For each of these phases it should be clear to a user what they can expect. In the case of an API, those expectations are:
+
+* **Alpha**
+   * API is not production ready yet and might contain bugs
+   * Recommended for non-business-critical use only because of potential for incompatible changes in subsequent releases
+   * May not be backwards-compatible with an API it intends to replace
+   * May not be highly performant or support all SDKs
+
+* **Beta**
+   * API is not production ready yet
```
Comment (artursouza) [inline guides/api-design.md:33]: Mention: API is intended to move to stable and not be deleted.

**Review by artursouza — APPROVED** [review]

File: guides/api-design.md (hunk 3 above)
Comment (johnewart) [reply guides/api-design.md:33]: Added

File: templates/lifecycle.md (hunk 2 above)
Comment (johnewart) [reply templates/lifecycle.md:51]: Added checklist item

File: templates/lifecycle.md (hunk 1 above)
Comment (johnewart) [reply templates/lifecycle.md]: I believe that would be the right place; I added a URL to point to there in this template so people know what it refers to (and to see existing information)

**Review by artursouza — APPROVED** [review]

**Pull request merged**